textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/queryrewriter/WordBase.java

```diff
@@ -0,0 +1,35 @@
+package edu.uci.ics.textdb.dataflow.queryrewriter;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.URL;
+import java.util.HashSet;
+
+/**
+ * Created by shiladityasen on 4/30/16.
+ * Class to provide simple abstraction of a dictionary of words with constant retrieval time
+ * Used by FuzzyTokenizer to process query
+ */
+public class WordBase
```

Comment:
Yeah. But we have to do a little refactoring. The following changes are needed.
1. Remove `WordBase` class.
2. Create one more constructor for `Dictionary` class to accept the `sourceFileName`.
3. Change the existing constructor `Dictionary(List<String>)` to accept `Collection<String>` and internally maintain the StringList as `Set` (so that `contains()` method performs in constant time).
4. `Dictionary` class will expose the methods `getNextValue()` and `contains()`.

@kishore-narendran .. If you agree, you can make these changes.